core/src/env/test_env.rs

```diff
@@ -428,38 +430,19 @@ impl TestEnv {
             .with(|test_env| test_env.borrow_mut().set_expected_return(expected_bytes))
     }
 
-    /// Sets the contract address for the next contract invocation.
-    pub fn set_address(new_address: srml::AccountId) {
-        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_address(new_address))
-    }
-
-    /// Sets the contract balance for the next contract invocation.
-    pub fn set_balance(new_balance: srml::Balance) {
-        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_balance(new_balance))
-    }
-
-    /// Sets the caller address for the next contract invocation.
-    pub fn set_caller(new_caller: srml::AccountId) {
-        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_caller(new_caller))
-    }
-
     /// Sets the input data for the next contract invocation.
     pub fn set_input(input_bytes: &[u8]) {
-        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_input(input_bytes))
-    }
-
-    /// Sets the random seed for the next contract invocation.
-    pub fn set_random_seed(random_seed_bytes: srml::Hash) {
-        TEST_ENV_DATA
-            .with(|test_env| test_env.borrow_mut().set_random_seed(random_seed_bytes))
+        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_input(input_bytes.to_vec()))
     }
 
-    /// Sets the timestamp for the next contract invocation.
-    pub fn set_now(timestamp: srml::Moment) {
-        TEST_ENV_DATA.with(|test_env| test_env.borrow_mut().set_now(timestamp))
-    }
+    impl_env_setters_for_test_env!(
+        (set_address, address, T::AccountId),
+        (set_balance, balance, T::Balance),
+        (set_caller, caller, T::AccountId),
+        (set_random_seed, random_seed, T::Hash),
+        (set_now, now, T::Moment)
+    );
 
-    /// Returns an iterator over all emitted events.
     pub fn emitted_events() -> impl IntoIterator<Item = Vec<u8>> {
```

Comment:
Why not return here the iterator directly?

Reply:
As far as I know/do, `IntoIterator` is better used for inputs, rather than outputs.